bin/main.rb

```diff
@@ -103,40 +113,66 @@ def display_board(cells)
 name1 = ui.prompt_player_name('player 1')
 symbol1 = 'X'
 array1 = []
-player1 = Player.new(name1, symbol1, array1)
+player1 = Player.new(name1.magenta, symbol1.magenta, array1)
 name2 = ui.prompt_player_name('player 2')
 symbol2 = 'O'
 array2 = []
-player2 = Player.new(name2, symbol2, array2)
-current_player = 'player1'
-
-board = Board.new(current_player)
-game_logic = GameLogic.new(player1, player2, board.cell)
-ui.display_board board.cell
+player2 = Player.new(name2.cyan, symbol2.cyan, array2)
+again="Y"
 
-until game_logic.game_end
+  while again == "Y"
+  current_player = 'player1'
+  board = Board.new(current_player)
+  game_logic = GameLogic.new(player1, player2, board.cell)
+  ui.display_board board.cell
+  moves = 0 
+  until game_logic.game_end || moves >8
+
+    mark = ui.get_mark(if current_player == 'player1'
+                        player1.name
+                      else
+                        player2.name
+                      end).to_i
```

Comment:
If you notice you are testing for player switching multiple times, if you handle the switching in the game logic you would not need to do that, you just use current_player to do whatever you want, knowing that current_player, is the player that is currently player at that point in time. Can we have a method that handles player switching in the game_logic?

Reply:
I only switched at the end of each turn, that part of the code doesn't check for a switch